.github/workflows/integration_tests.yml

```diff
@@ -23,28 +29,166 @@ on:
+      image: ${{ inputs.cudaq_test_image }}
       options: --user root
-
+    outputs:
+      cudaq_commit: ${{ steps.commit-sha.outputs.sha }}
     steps:
       - name: Get commit SHA
         id: commit-sha
         run: |
-          echo "sha=$(cat $CUDA_QUANTUM_PATH/build_info.txt | grep -o 'source-sha: \S*' | cut -d ' ' -f 2)" >> $GITHUB_OUTPUT
+          if [ -n "${{ inputs.commit_sha }}" ]; then
+            echo "sha=${{ inputs.commit_sha }}" >> $GITHUB_OUTPUT
+          else
+            echo "sha=$(cat $CUDA_QUANTUM_PATH/build_info.txt | grep -o 'source-sha: \S*' | cut -d ' ' -f 2)" >> $GITHUB_OUTPUT
+          fi
+  
+  build_nvcf_image:
+    name: Build NVCF deployment image
+    runs-on: ubuntu-latest
+    needs: metadata
+    environment: backend-validation
+    if: (inputs.target == 'nvcf' || github.event_name == 'schedule' || inputs.target == 'nightly')
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ needs.metadata.outputs.cudaq_commit }}
+          fetch-depth: 1
+
+      - name: Set up context for buildx
+        run: |
+          docker context create builder_context
+
+      - name: Set up buildx runner
+        uses: docker/setup-buildx-action@v3
+        with:
+          endpoint: builder_context
+
+      - name: Login to NGC container registry
+        uses: docker/login-action@v3
+        with:
+          registry: nvcr.io
+          username: '$oauthtoken'
+          password: ${{ secrets.NGC_NVCF_CREDENTIALS }}
+
+      - name: Build NVCF image
+        id: docker_build
+        uses: docker/build-push-action@v5
+        with:
+          context: .
+          file: ./docker/release/cudaq.nvcf.Dockerfile
+          build-args: |
+            base_image=${{ inputs.cudaq_nvcf_deploy_image }}
+          tags: nvcr.io/${{ secrets.NGC_NVCF_ORG }}/cuda-quantum/cuda-quantum:nightly
+          platforms: linux/amd64
+          provenance: false
+          push: true
+  
+  deploy_nvcf_test_function:
+    name: Deploy NVCF function
+    runs-on: ubuntu-latest
+    needs: [metadata, build_nvcf_image]
+    if: (inputs.target == 'nvcf' || github.event_name == 'schedule' || inputs.target == 'nightly')
+    environment: backend-validation
+    outputs:
+      nvcf_function_version_id: ${{ steps.deploy.outputs.nvcf_function_version_id }}
+    
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ needs.metadata.outputs.cudaq_commit }}
+          fetch-depth: 1
+
+      - name: Install NGC CLI
+        uses: ./.github/actions/install-ngc-cli
+        with:
+          version: 3.38.0
+          checksum: 427c67684d792b673b63882a6d0cbb8777815095c0f2f31559c1570a91187388
+
+      - name: Deploy NVCF Function
+        id: deploy
+        env:
+          NGC_CLI_API_KEY: ${{ secrets.NGC_NVCF_CREDENTIALS }}
+          NGC_CLI_ORG: ${{ secrets.NGC_NVCF_ORG }}
```

Comment:
can't you use the org name instead of the org ID here? It's odd that the org is a variable but the team is hardcoded. I would say we either hardcode both or I create a normal variable for the org name and for the team name.

Reply:
No, I cannot use the org name. I've hardcoded the org Id instead to keep it consistent.